source/slang/slang.cpp

```diff
@@ -1375,8 +1394,10 @@ void FrontEndCompileRequest::addTranslationUnitSourceFile(
     // paths were not taken into account by this function.
     //
 
+    PathInfo pathInfo;
+
     ComPtr<ISlangBlob> sourceBlob;
-    SlangResult result = loadFile(path, sourceBlob.writeRef());
+    SlangResult result = loadFile(path, pathInfo, sourceBlob.writeRef());
```

Comment:
It seems like things are at this point a bit roundabout, where we have the file-system layer that is doing all this caching and unique-ing and so forth, and then we have the "source manager" layer, which nominally owns the unique/deduped source files. It seems like we eventually want a single routine that turns a path into a `SourceFile`, and handles all the details of what that entails, whereas right now that logic is here in `addTranslationUnitSourceFile`, where is it feels like it is mixed up with higher-level concerns of the `FrontEndCompileRequest`.

That isn't a condemnation of this PR, but rather a comment that the infrastructure has progressed far enough that some of the use-site code like this is looking a bit silly.

Reply:
Not sure I'd describe this as 'silly' - in so far as we get two things now. The contents of the source file and its 'identity' the PathInfo. Both are appropriate/useful/needed, in what we have.

That there could be some other mechanism such that 'identity' was a pointer to something. But that introduces other issues. Well what is the thing - it's probably not 'source'. I also can't easily serialize or show it's identity. Not a show stopper I'll grant you.